.github/workflows/pr-workflow.yml

```
@@ -0,0 +1,25 @@
+name: PR build
+
+on: [pull_request]
```

This might pose a problem as GitHub Actions workflows triggered from forks do not have access to secrets: https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/#improvements-for-public-repository-forks
Solution:
```suggestion
on: [pull_request_target]
```

---

I don't think this will work because this runs in the context of the base branch - it's only used for labelling, commenting, etc. not running code from an untrusted fork.

---

`pull_request_target` doesn't use the pull request code though from what i've read, so it doesn't open up security issues. i'm hoping that github actions run on servers that have no pull rate limiting :crossed_fingers:

---

https://github.com/actions/virtual-environments/issues/1445 explains that GitHub actions have no pull limit (agreement with docker hub) so no need to login.